File: datavar.r
# Integer
my_int <- 42
cat("Integer:", my_int, "\n")

# Float
my_float <- 3.14
cat("Float:", my_float, "\n")

# Double
my_double <- 3.14159
cat("Double:", my_double, "\n")

# Character
my_char <- "A"
cat("Char:", my_char, "\n")

# String
my_string <- "Hello, R!"
cat("String:", my_string, "\n")

# Boolean
my_boolean <- TRUE
cat("Boolean:", my_boolean, "\n")

# Array
my_array <- c(1, 2, 3, 4, 5)
cat("Array:", my_array, "\n")

# List
my_list <- list(1, 2, 3, 4, 5)
cat("List:", my_list, "\n")

# Named List (equivalent to a dictionary or map)
my_map <- list(one=1, two=2, three=3)
cat("Map:\n")
for (name in names(my_map)) {
  cat(name, "=>", my_map[[name]], "\n")
}

# Function
my_function <- function(x) {
  return(x * 2)
}
cat("Function (2 * 3):", my_function(3), "\n")

# Data Frame (Struct equivalent)
my_person <- data.frame(name="Alice", age=25)
cat("Struct: Name =", my_person$name, ", Age =", my_person$age, "\n")

# NULL
my_null <- NULL
cat("NULL:", my_null, "\n")

File: datavar.kt
fun main() {
    // Integer
    val myInt: Int = 42
    println("Integer: $myInt")

    // Float
    val myFloat: Float = 3.14F
    println("Float: $myFloat")

    // Double
    val myDouble: Double = 3.14159
    println("Double: $myDouble")

    // Char
    val myChar: Char = 'A'
    println("Char: $myChar")

    // String
    val myString: String = "Hello, Kotlin!"
    println("String: $myString")

    // Boolean
    val myBoolean: Boolean = true
    println("Boolean: $myBoolean")

    // Array
    val myArray: Array<Int> = arrayOf(1, 2, 3, 4, 5)
    println("Array: ${myArray.joinToString(", ")}")

    // List
    val myList: List<Int> = listOf(1, 2, 3, 4, 5)
    println("List: ${myList.joinToString(", ")}")

    // Set
    val mySet: Set<Int> = setOf(1, 2, 3, 4, 5)
    println("Set: ${mySet.joinToString(", ")}")

    // Map
    val myMap: Map<String, Int> = mapOf("one" to 1, "two" to 2, "three" to 3)
    println("Map: $myMap")

    // Pair (Tuple)
    val myPair: Pair<Int, String> = Pair(42, "Kotlin")
    println("Pair: $myPair")

    // Function
    val myFunction: (Int) -> Int = { x -> x * 2 }
    println("Function (2 * 3): ${myFunction(3)}")

    // Data class (Struct equivalent)
    data class Person(val name: String, val age: Int)
    val myPerson = Person("Alice", 25)
    println("Data class: Name = ${myPerson.name}, Age = ${myPerson.age}")

    // Null
    val myNull: String? = null
    println("Null: $myNull")
}
